src/main/scala/no/gdl/bookapi/repository/FeaturedContentRepository.scala

```diff
@@ -0,0 +1,99 @@
+/*
+ * Part of GDL book_api.
+ * Copyright (C) 2018 Global Digital Library
+ *
+ * See LICENSE
+ */
+
+package no.gdl.bookapi.repository
+
+import io.digitallibrary.language.model.LanguageTag
+import no.gdl.bookapi.model.api
+import no.gdl.bookapi.model.api.OptimisticLockException
+import no.gdl.bookapi.model.api.internal.FeaturedContentId
+import no.gdl.bookapi.model.domain.FeaturedContent
+import scalikejdbc._
+
+import scala.util.{Success, Try}
+
+
+trait FeaturedContentRepository {
+  val featuredContentRepository: FeaturedContentRepository
+
+  class FeaturedContentRepository {
+
+    def updateContent(featuredContent: api.FeaturedContent)(implicit session: DBSession = AutoSession): Try[FeaturedContentId] = {
+      val f = FeaturedContent.column
+      val newRevision = featuredContent.revision + 1
+      Try {
+        val affectedRows = update(FeaturedContent).set(
+          f.revision -> newRevision,
+          f.language -> featuredContent.language.code,
+          f.title -> featuredContent.title,
+          f.description -> featuredContent.description,
+          f.link -> featuredContent.link,
+          f.imageUrl -> featuredContent.imageUrl)
+          .where.eq(f.id, featuredContent.id)
+          .toSQL.update().apply()
+
+        if (affectedRows != 1) {
+          throw new OptimisticLockException()
+        } else {
+          FeaturedContentId(featuredContent.id)
+        }
+      }
+    }
+
+    def deleteContent(id: Long)(implicit session: DBSession = AutoSession): Try[Unit] = {
+      val affectedRows = withSQL {
+        delete
+          .from(FeaturedContent as f)
+          .where.eq(f.id, id)
+      }
+        .update()
+        .apply()
+      if (affectedRows != 1) {
+        throw new OptimisticLockException()
+      } else {
+        Success()
+      }
+    }
+
+    private val f = FeaturedContent.syntax
+
+    def addContent(featuredContent: FeaturedContent)(implicit session: DBSession = AutoSession): FeaturedContent = {
+
+      val f = FeaturedContent.column
+      val startRevision = 1
+
+      val id =
+        sql"""
+        insert into ${FeaturedContent.table} (
```

Comment:
Her ville jeg foretrekke at vi brukte den andre formen for insert støttet av scalikejdbc:

Eksempel:
```
val id = insert.into(Chapter).namedValues(
        ch.translationId -> newChapter.translationId,
        ch.seqNo -> newChapter.seqNo,
        ch.title -> newChapter.title,
        ch.content -> newChapter.content
      ).toSQL.updateAndReturnGeneratedKey().apply()
```

Den formen som du har benyttet er ikke nødvendig her. Såvidt jeg har forstått, så trenger man å benytte den ved f.eks insert av arrays